string/string.cc

```diff
@@ -1,12 +1,113 @@
 #include "string.h"
 
-String::String()
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <string>
+
+
+const char* EMPTY = "";
+
+
+size_t strlen(const char* s)
+{
+	size_t len = 0;
+
+	while (s[len] != '\0')
+		len++;
+
+	return len;
+}
+
+
+String::String() {}
+
+String::~String() {}
+
+String::String(const String& o)
+{
+	mSize = o.size();
+
+	mValue = char_ptr(new char[mSize + 1]);
+
+	std::copy_n(o.begin(), mSize, begin());
+
+	mValue[mSize + 1] = '\0';
```

Comment:
You may just copy mSize+1 chars. Yoyu are copying from your own String, so you know that it contains the null-terminator.